### ISSUE TITLE
[QUESTION] Compare contrast with JS libraries

### ISSUE THREAD

**Issue by mwpowellhtx** [mono/SkiaSharp#1847]
Q: How does _Skia(Sharp)_ compare contrast with JS libraries like [_Babylon_](https://babylonjs.com) of [_THREE_](https://threejs.org)? Aside from the obvious, I get it, the `CSharp` interop aspects. Also, that the JS libraries are purely JS canvas oriented. But perhaps in a bit more depth, insightfulness. Thank you...

**Comment by artemiusgreat** [comment]
My knowledge of Skia is limited, so take it with a grain of salt. 

Blazor bindings of SkiaSharp in Web work with HTML canvas element. 
This element provides two types of API - 2D Canvas and 3D Web GL. 
https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext

```
const canvas = document.querySelector("canvas");
const c2d = canvas.getContext("2d");
const c3d = canvas.getContext("webgl");
```

The same APIs are already exposed in SkiaSharp - SKCanvasView and SKGLView. 
https://github.com/mono/SkiaSharp/tree/main/source/SkiaSharp.Views.Blazor/SkiaSharp.Views.Blazor 

At the same time, Skia positions itself as a 2D framework, so 3D Web GL in Skia may not be as evolved as 2D Canvas. 
In other words, rendering something 3D in Skia may involve more coding compared to JS frameworks above, but overal possibilities and performance should be the same.

**Comment by mattleibow** [comment]
@artemiusgreat is correct. Skia is only really a 2D graphics engine. It can draw on WebGL for hardware acceleration, but still only 2D things.


For 3D, you will need to use another library. I am not sure of any offhand.

**Comment by mwpowellhtx** [comment]
> For 3D, you will need to use another library. I am not sure of any offhand.

Well, that was partly my question, along the lines of, what is _Skia_ and/or _SkiaSharp_ bringing to the table, exactly. Sounds like they are completely different target markets, this, compared contrasted with a _THREE_ or a _BabylonJS_, for instance.
